docs/spec.md

```diff
@@ -4790,22 +5122,25 @@ where `pred_val = rank(pred) == 0 ? pred : pred[i0, ..., iR-1]`.
 
 #### Inputs
 
-| Label | Name       | Type                | Constraints |
-|-------|------------|---------------------|-------------|
-| (I1)  | `pred`     | tensor of type `i1` | (C1)        |
-| (I2)  | `on_true`  | tensor              | (C1), (C2)  |
-| (I3)  | `on_false` | tensor              | (C2)        |
+| Label | Name       | Type                       | Constraints |
+|-------|------------|----------------------------|-------------|
+| (I1)  | `pred`     | tensor of type `i1`        | (C1)        |
+| (I2)  | `on_true`  | tensor or quantized tensor | (C1-C4)     |
+| (I3)  | `on_false` | tensor or quantized tensor | (C2-C3)     |
 
 #### Outputs
 
-| Name     | Type   | Constraints |
-|----------|--------|-------------|
-| `result` | tensor | (C2)        |
+| Name     | Type                       | Constraints |
+|----------|----------------------------|-------------|
+| `result` | tensor or quantized tensor | (C2)        |
 
 #### Constraints
 
 * (C1) Either `rank(pred)` $=$ `0` or `shape(pred)` $=$ `shape(on_true)`.
 * (C2) `on_true`, `on_false` and `result` have same type.
+* If the operation uses quantized tensors:
+  * (C3) `is_quantized_tensor(on_true) and is_quantized_tensor(on_false)`.
+  * (C4) `is_empty(quantization_dimension(on_true))`.
```

Comment:
Any particular reason to treat SelectOp differently from ClampOp?

Reply:
As the op selects one or the other operand based on `pred` value, I thought a simplifiled semantics can be proposed. 

However, based on 
>> This way, we can describe the semantics of elementwise ops with respect to quantization as "we treat quantized tensors as floating-point tensors and wrap floating-point computations in dequantize/quantize".

I agree that it would be better to have a common theme around elementwise ops.
Towards that I have updated the `select` op sepc. PTAL. 
I have added a new quantization computation `dequantize_op_quantize_select` to accommodate the boolean `pred` operand. I am not very excited about introducing a new function for an op. Even if represent `pred` as a quantized tensor, so as to use the original `dequantize_op_quantize`, I have to convert the dequantized floating-point `pred` to boolean type before applying the operation. 

Let me know what do you think.